### PR TITLE
Add APF Testing to Prow CI

### PR DESCRIPTION
**Pull request by fbledsoe**
/hold

Integrating new regression testing for custom FlowSchemas and PriorityLevelConfigurations into Prow. Test will verify that new API Priority and Fairness can be properly added, and that excessive requests are properly handled.

I initially created the custom-flow-control test in the svt repository, but just ended up pasting it into the commands.sh file per Liquan's request.

### CODE REVIEW

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-apf.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      product: ocp
+      stream: nightly
+      version: "4.15"
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: run-api-apf-customized-flowcontrol
```
Comment (paigerube14) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-apf.yaml:24]: maybe lets just add this test as an item in the test list in the aws-4.15nightly file

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-apf-run-api-apf-customized-flowcontrol

**Comment by liqcui** [comment]
> @liqcui Can we run this test as e2e test like https://github.com/openshift/openshift-tests-private/blob/master/test/extended/perfscale/ocperf.go? Create a new cluster just for this single test is not quite efficient.

@svetsa-rh any comments?  Is that possible add `chainopenshift-qe-run-api-apf-customized-flowcontrol` to after line#47 https://github.com/openshift/release/blob/master/ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86.yaml#L47 and test with controlpance together, if anything failure about this chain, dev team can reach us. so we can share cluster to save cost, thanks!

**Comment by fbledsoe** [comment]
/pj-rehearse 
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by svetsa-rh** [comment]
> > @liqcui Can we run this test as e2e test like https://github.com/openshift/openshift-tests-private/blob/master/test/extended/perfscale/ocperf.go? Create a new cluster just for this single test is not quite efficient.
> 
> @svetsa-rh any comments? Is that possible add `chainopenshift-qe-run-api-apf-customized-flowcontrol` to after line#47 https://github.com/openshift/release/blob/master/ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86.yaml#L47 and test with controlpance together, if anything failure about this chain, dev team can reach us. so we can share cluster to save cost, thanks!

@liqcui @qiliRedHat we could except it doesn't make sense to bundle it with other workload tests as they are totally unrelated workloads. 
I am not entirely convinced about bundling it with other tests, but if we really want to do it to save some money, may be we can create a new chain to run this after loaded upgrade tests and see how the experiment goes. Since these 2 tests are specific to QA, we can avoid confusion to others.
And eventually as we add more perf regression tests to Prow, we can couple with other regression tests and separate from loaded upgrade.

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by svetsa-rh** [comment]
> > > > @liqcui Can we run this test as e2e test like https://github.com/openshift/openshift-tests-private/blob/master/test/extended/perfscale/ocperf.go? Create a new cluster just for this single test is not quite efficient.
> > > 
> > > 
> > > @svetsa-rh any comments? Is that possible add `chainopenshift-qe-run-api-apf-customized-flowcontrol` to after line#47 https://github.com/openshift/release/blob/master/ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86.yaml#L47 and test with controlpance together, if anything failure about this chain, dev team can reach us. so we can share cluster to save cost, thanks!
> > 
> > 
> > @liqcui @qiliRedHat we could except it doesn't make sense to bundle it with other workload tests as they are totally unrelated workloads. I am not entirely convinced about bundling it with other tests, but if we really want to do it to save some money, may be we can create a new chain to run this after loaded upgrade tests and see how the experiment goes. Since these 2 tests are specific to QA, we can avoid confusion to others. And eventually as we add more perf regression tests to Prow, we can couple with other regression tests and separate from loaded upgrade.
> 
> @fbledsoe @svetsa-rh @qiliRedHat Let's add the new chain to after line46 https://github.com/openshift/release/blob/master/ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-loaded-upgrade-from-4.14.yaml#L46 You need to increase more replicas for the openshift-qe-run-api-apf-customized-flowcontrol to get expected result, thanks!

@liqcui 
> You need to increase more replicas for the openshift-qe-run-api-apf-customized-flowcontrol to get expected result, thanks!
What is the proposed increase of number of nodes? Let us weigh in pros vs cons before making the change.

**Comment by liqcui** [comment]
> > > > > @liqcui Can we run this test as e2e test like https://github.com/openshift/openshift-tests-private/blob/master/test/extended/perfscale/ocperf.go? Create a new cluster just for this single test is not quite efficient.
> > > > 
> > > > 
> > > > @svetsa-rh any comments? Is that possible add `chainopenshift-qe-run-api-apf-customized-flowcontrol` to after line#47 https://github.com/openshift/release/blob/master/ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86.yaml#L47 and test with controlpance together, if anything failure about this chain, dev team can reach us. so we can share cluster to save cost, thanks!
> > > 
> > > 
> > > @liqcui @qiliRedHat we could except it doesn't make sense to bundle it with other workload tests as they are totally unrelated workloads. I am not entirely convinced about bundling it with other tests, but if we really want to do it to save some money, may be we can create a new chain to run this after loaded upgrade tests and see how the experiment goes. Since these 2 tests are specific to QA, we can avoid confusion to others. And eventually as we add more perf regression tests to Prow, we can couple with other regression tests and separate from loaded upgrade.
> > 
> > 
> > @fbledsoe @svetsa-rh @qiliRedHat Let's add the new chain to after line46 https://github.com/openshift/release/blob/master/ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.15-nightly-x86-loaded-upgrade-from-4.14.yaml#L46 You need to increase more replicas for the openshift-qe-run-api-apf-customized-flowcontrol to get expected result, thanks!
> 
> @liqcui
> 
> > You need to increase more replicas for the openshift-qe-run-api-apf-customized-flowcontrol to get expected result, thanks!
> 
> What is the proposed increase of number of nodes? Let us weigh in pros vs cons before making the change.

@svetsa-rh It's not worker node's replicas, it's APF test pod's replicas

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by liqcui** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse
periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-run-api-apf-customized-flowcontrol

**Comment by fbledsoe** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-24nodes-loaded-upgrade-414to415

**Comment by fbledsoe** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-24nodes-loaded-upgrade-414to415

**Comment by fbledsoe** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-24nodes-loaded-upgrade-414to415

**Comment by fbledsoe** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-24nodes-loaded-upgrade-414to415

**Comment by fbledsoe** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-24nodes-loaded-upgrade-414to415

**Comment by liqcui** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-loaded-upgrade-from-4.14-perfscale-aws-24nodes-loaded-upgrade-414to415

File: ci-operator/step-registry/openshift-qe/run-api-apf-customized-flowcontrol/openshift-qe-run-api-apf-customized-flowcontrol-commands.sh [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+check_errors
+
+echo -e "\n======Starting cleanup======"
+
+delete_flow_schema
+
+delete_priority_level_configuration
+
+delete_controller
```
Comment (liqcui) [inline ci-operator/step-registry/openshift-qe/run-api-apf-customized-flowcontrol/openshift-qe-run-api-apf-customized-flowcontrol-commands.sh:303]: please move delete_controller to L298

**Comment by svetsa-rh** [comment]
@liqcui Should we need to keep this PR frozen and open or do we want to close it or let it become stale?

**Comment by liqcui** [comment]
@svetsa-rh Let's close this one, I will re-open a new new today, thanks!

**Pull request closed without merging**